### ISSUE TITLE
Question regarding fine tuning

### ISSUE THREAD

**Issue by MichaelMMeskhi** [hila-chefer/Transformer-Explainability#40]
I am trying to fine tune on a custom image dataset. Using base 224 patch 16, I am setting the all params to false and changing the `head` output dims to 2 (binary classification). When trying to train this model, I get the error `"cannot set a hook on a tensor that does not require grads".` 

I simply added a conditional statement in `vit_lrp.py`:

```
if self.train and x.requires_grad:
    x.register_hook()

if self.train and attn.requires_grad:
    attn.register_hook()
```

Just making sure this isn't breaking anything as this does fix my issue to fine tune.

**Comment by hila-chefer** [comment]
Hi @MichaelMMeskhi, thanks for your interest!
The second hook you attached is originally used to calculate the gradients of each attention head (as our method averages across the heads using each head's gradient as weights).
By not collecting the gradients, the method will not work (I think it will return None as the gradient information and then you should see an exception).
However, since you do not need the relevance maps when training fine-tuning, this should have no effect on your visualization results after completing the fine-tuning. In fact, you can even fine-tune your model using the simple no LRP code as appears in [the original implementation](https://github.com/rwightman/pytorch-image-models) and then just save your weights and [replace the url link in our code for visualization](https://github.com/hila-chefer/Transformer-Explainability/blob/main/baselines/ViT/ViT_LRP.py#L29) with your checkpoint.

I hope this helps, but please let me know if you have any follow-up questions.

Best,
Hila.

**Comment by MichaelMMeskhi** [comment]
Thank you for your reply @hila-chefer! Yes, I did this originally for fine-tuning. Thank you, I managed to get things working.

Best,
Michael
